src/tid.rs

```diff
@@ -169,7 +169,7 @@ impl Registration {
 // panics. T_T
 // Just skip TID reuse and use loom's lazy_static macro to ensure we have a
 // clean initial TID on every iteration, instead.
-#[cfg(not(loom))]
+#[cfg(not(all(test, loom)))]
```

Comment:
I think this should be
```suggestion
#[cfg(not(all(loom, any(feature = "loom", test)))]
```